### PR TITLE
feat(Cost Center): Distributed Cost Center

### PR DESCRIPTION
**Pull request by karthikeyan5**
This feature enhances the Cost Center dimension to allow the distribution of the tagged value to a set of Cost Centers.

A distributed cost center can be created out of different ratios of ordinary cost centers and it can be tagged against the GL entries.

Financial statement reports are also accordingly modified when cost center filter is added.

**Screenshot of Cost Center with Distributed Cost Center**
image

### CODE REVIEW

**Comment by sammish** [comment]
@nabinhait Thanks

**Review by nabinhait — CHANGES_REQUESTED** [review]
- The actual amount column in Budget Variance Report should also be updated based on distribution cost center.
- docs is pending

Otherwise, the PR is good to go.

**Comment by KaviyaPeriyasamy** [comment]
> * The actual amount column in Budget Variance Report should also be updated based on distribution cost center.
> * docs is pending
> 
> Otherwise, the PR is good to go.

Hello @nabinhait ,
This https://github.com/frappe/erpnext/pull/21531/commits/84f863fdee2af9afece0f6a257e03806d444fb45 commit will update the actual amount column based on the distributed cost center.

**Comment by KaviyaPeriyasamy** [comment]
> @KaviyaPeriyasamy thanks for the dedicated support.

It's my pleasure.

**Comment by KaviyaPeriyasamy** [comment]
@nabinhait 
Corresponding docs were updated in this PR https://github.com/frappe/erpnext_documentation/pull/85

File: erpnext/accounts/report/general_ledger/general_ledger.py [hunk 1]
```diff
@@ -128,6 +128,40 @@ def get_gl_entries(filters):
...
+			party_type,
+			party,
+			voucher_type,
+			voucher_no,
+			cost_center, project,
+			against_voucher_type,
+			against_voucher,
+			account_currency,
+			remarks, against, 
+			is_opening {select_fields_with_percentage}
```
Comment (nabinhait) [inline erpnext/accounts/report/general_ledger/general_ledger.py]:
```suggestion
			is_opening, `tabGL Entry`.creation {select_fields_with_percentage}
```

File: erpnext/accounts/report/general_ledger/general_ledger.py [hunk 2]
```diff
@@ -137,9 +171,10 @@ def get_gl_entries(filters):
 			remarks, against, is_opening {select_fields}
```
Comment (nabinhait) [inline erpnext/accounts/report/general_ledger/general_ledger.py]:
```suggestion
			remarks, against, is_opening, creation {select_fields}
```

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by KaviyaPeriyasamy** [comment]
@nabinhait 
Suggested changes were applied.

**Review by nabinhait — APPROVED** [review]

**Comment by sammish** [comment]
@nabinhait thanks for your great support 

@karthikeyan5 @Vigneshsekar @KaviyaPeriyasamy thanks team for the unlimited support.

**Comment by deepeshgarg007** [comment]
@Mergifyio backport version-13-beta-pre-release

**Pull request merged**